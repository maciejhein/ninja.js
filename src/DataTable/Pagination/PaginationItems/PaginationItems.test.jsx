import React from 'react';
import { mount, shallow } from 'enzyme';

import PaginationItem from './PaginationItems';

describe('PaginationItem', () => {
  let props; 

  beforeEach(() => {
    props = {
      pageNumber: 0,
      isActive: false,
      onChange: jest.fn(),
    };
  });

  it('should renders correctly ', () => {
    const component = shallow(<PaginationItem {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should calls `onChange` function when button clicked', () => {
    const component = mount(<PaginationItem {...props} />);

    component.find('button').simulate('click');

    expect(props.onChange).toHaveBeenCalled(); 
  });

  it('should render button with class `button-outline` when page is active', () => {
    const newProps = {
      ...props,
      isActive: true
    }
    const component = mount(<PaginationItem {...newProps} />);

    expect(component.find('button').hasClass('button-outline')).toBe(true);
  });
});