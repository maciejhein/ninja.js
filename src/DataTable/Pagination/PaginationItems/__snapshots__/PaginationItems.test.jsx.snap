// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationItem renders correctly  1`] = `
<li
  className="page-item mr-1"
>
  <button
    className="page-link "
    onClick={[Function]}
  >
    1
  </button>
</li>
`;
